File: electric_index/lib/models/meter.dart
class Meter {
  final String meterId;
  final int companyId;
  final String? meterName;
  final String? imageMeter;

  Meter({
    required this.meterId,
    required this.companyId,
    this.meterName,
    this.imageMeter,
  });

  factory Meter.fromMap(Map<String, dynamic> map) {
    return Meter(
      meterId: map['meter_id'],
      companyId: map['company_id'],
      meterName: map['meter_name'],
      imageMeter: map['image_meter'],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'meter_id': meterId,
      'company_id': companyId,
      'meter_name': meterName,
      'image_meter': imageMeter,
    };
  }
}

File: Index_Electric_server/src/main/java/com/index_electric_server/reading/entity/Meter.java
package com.index_electric_server.reading.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(
        name = "meters",
        indexes = {
                @Index(name = "ix_meter_company", columnList = "company_id"),
                @Index(name = "ix_meter_location", columnList = "location_id")
        }
)
public class Meter {

    @Id
    @Column(name = "meter_id", length = 64)
    private String meterId;

    @Column(name = "company_id")
    private Long companyId;

    @Column(name = "meter_name")
    private String meterName;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    public Meter() {}

    @PrePersist
    public void prePersist() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }

    public String getMeterId() {
        return meterId;
    }

    public void setMeterId(String meterId) {
        this.meterId = meterId;
    }

    public Long getCompanyId() {
        return companyId;
    }

    public void setCompanyId(Long companyId) {
        this.companyId = companyId;
    }

    public String getMeterName() {
        return meterName;
    }

    public void setMeterName(String meterName) {
        this.meterName = meterName;
    }

    public Long getLocationId() {
        return locationId;
    }

    public void setLocationId(Long locationId) {
        this.locationId = locationId;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
}